crates/uv-resolver/src/marker.rs

```diff
@@ -94,62 +92,84 @@ fn string_is_disjoint(this: &MarkerExpression, other: &MarkerExpression) -> bool
 ///
 /// This is useful in cases where creating conjunctions or disjunctions might occur in a non-deterministic
 /// order. This routine will attempt to erase the distinction created by such a construction.
-pub(crate) fn normalize(tree: &mut MarkerTree) {
+pub(crate) fn normalize(tree: MarkerTree) -> Option<MarkerTree> {
     match tree {
-        MarkerTree::And(trees) | MarkerTree::Or(trees) => {
```

Comment:
I split these into separate branches. There's a little bit of duplication, but it makes things easier with ownership (and the branches now have more deviations than before between the two cases).